pages/data/stations/_id.vue

```diff
@@ -213,9 +283,136 @@ export default {
     this.$store.dispatch('stations/downloadStations')
       .then(() => {
         this.populate()
+        // this.refreshMetrics()
       })
+    Promise.all([
+      this.$store.dispatch('stations/downloadStationsByDataset'),
+    ]).then(() => {
```

Comment:
`this.$store.dispatch('stations/downloadStationsByDataset')` already returns a Promise from the 2 different httpRequest that it makes, so you don't actually need the `Promise.all`. You can simply link the `.then()` success handler directly.

```suggestion
    this.$store.dispatch('stations/downloadStationsByDataset')
      .then(() => {
```